main/Main.hs

```diff
@@ -46,7 +47,9 @@ main = do
   hSetBuffering stdin  LineBuffering
   hSetBuffering stderr NoBuffering
 
-  conf <- readOptions
+  conf' <- readOptions
+  host  <- getHostName
```

Comment:
The name that is resolved is rarely the public name for your api so probably better to have a config option with a default to 'localhost' ... i.e. no Network.BSD (might even break on windows)